### PR TITLE
Add upgrade notice document

### PR DESCRIPTION
**Pull request by lonegunmanb**
## Issue number

#000

## Checklist before requesting a review
- [ ] The pr title can be used to describe what this pr did in `CHANGELOG.md` file
- [ ] I have executed pre-commit on my machine
- [ ] I have passed pr-check on my machine

Thanks for your cooperation!

### CODE REVIEW

**Comment by lonegunmanb** [comment]
Hi @zioproto I've added an upgrade notice document for incoming `v8`, would you please give it a review? Thanks!

File: NoticeOnUpgradeTov8.0.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+2. The `create_before_destroy` attribute is added to the `node_pools` variable as an object field. This attribute determines whether a new node pool should be created before the old one is destroyed during updates. By default, it is set to `true`.
+
+3. The naming of extra node pools has been updated. Now, a random UUID is used as the seed for the random suffix in the name of the node pool, instead of the JSON-encoded value of the node pool. **This naming suffix only apply for extra node pools that create before destroy.**
+
+You're recommended to set `var.cluster_name_random_suffix` to `true` explicitly, and you'll see a random suffix in your cluster's name. If you don't like this suffix, please remember now a new cluster with the same name would be created before the old one has been deleted. If you do want to recreate the cluster, please run `terraform destroy` first.
```
Comment (zioproto) [inline NoticeOnUpgradeTov8.0.md:11]: Suggestion: 

If you don't like this suffix, please remember that you can't create a new cluster with the same name before the old one has been deleted. If you do want to recreate the cluster, please run `terraform destroy` first.

File: NoticeOnUpgradeTov8.0.md [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+1. The variable `http_application_routing_enabled` has been removed from the module. This variable was previously used to enable HTTP Application Routing Addon.
+
+2. The `http_application_routing_enabled` output has been removed from `outputs.tf`. This output was previously used to display whether HTTP Application Routing was enabled.
+
+3. The `http_application_routing_enabled` attribute has been removed from the `azurerm_kubernetes_cluster` resource in `main.tf`. This attribute was previously used to enable HTTP Application Routing for the Kubernetes cluster.
+
+4. The `http_application_routing_enabled` attribute has been added to the `ignore_changes` lifecycle block of the `azurerm_kubernetes_cluster` resource in `main.tf`. This means changes to this attribute will not trigger the resource to be updated.
+
+These changes mean that users of this module will no longer be able to enable HTTP Application Routing through this module.
```
Comment (zioproto) [inline NoticeOnUpgradeTov8.0.md:25]: Suggestion:

These changes mean that users of this module will no longer be able to enable HTTP Application Routing.

File: NoticeOnUpgradeTov8.0.md [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+The new feature for the Ingress in AKS is [Managed NGINX ingress with the application routing add-on](https://learn.microsoft.com/en-us/azure/aks/app-routing?tabs=default%2Cdeploy-app-default), you can enable this with `var.web_app_routing`.
```
Comment (zioproto) [inline NoticeOnUpgradeTov8.0.md:27]: Suggestion:

The new feature for the in-cluster ingress in AKS is ...

File: NoticeOnUpgradeTov8.0.md [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+
+3. `create_role_assignments_for_application_gateway`: This is a boolean variable that determines whether to create the corresponding role assignments for the application gateway or not. By default, it is set to `true`. Role assignments are necessary for the Application Gateway to function correctly with the AKS cluster. If set to `true`, the module will create the necessary role assignments on the Application Gateway.
+
+## Redesign variable for new created `azurerm_log_analytics_workspace`
+
+1. The `cluster_log_analytics_workspace_name` variable has been replaced with a new `new_log_analytics_workspace` variable. This new variable is an object that allows for more granular configuration of the Log Analytics Workspace.
```
Comment (zioproto) [inline NoticeOnUpgradeTov8.0.md:57]: Is this still under development ? Correct ? https://github.com/Azure/terraform-azurerm-aks/pull/515

**Review by zioproto — COMMENTED** [review]
Minor comments. Please check if the section about log analytics needs to be changed. Thanks

File: NoticeOnUpgradeTov8.0.md (hunk 4 above)
Comment (lonegunmanb) [reply NoticeOnUpgradeTov8.0.md:57]: You're correct, if we decide to not merge #515 then I'll remove this section.

**Comment by lonegunmanb** [comment]
Thanks @zioproto for the code review, I've fixed all comments you've left but only one left, I'll decide how should we process this log analysis section once we're on the same page on #515 .

**Pull request merged**